src/Cli/dotnet/Commands/Run/RunCommand.cs

```diff
@@ -742,4 +786,7 @@ public static ParseResult ModifyParseResultForShorthandProjectOption(ParseResult
         var newParseResult = Parser.Parse(tokensToParse);
         return newParseResult;
     }
+
+    [GeneratedRegex(@"\$\((?<token>[^\)]+)\)", RegexOptions.IgnoreCase, "en-US")]
+    private static partial Regex MSBuildPropertyUsagePattern();
```

Comment:
Consider using a partial property instead:
```suggestion
    private static partial Regex MSBuildPropertyUsagePattern { get; }
```